### PR TITLE
Rel 507447 update workspace helpers to rest

### PR DESCRIPTION
**Pull request by aarongilbert123**

### CODE REVIEW

**Review by TommyBigNose — DISMISSED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/WorkspaceHelper.cs [hunk 1]
```diff
@@ -235,41 +296,149 @@ public async Task<int> GetFirstWorkspaceArtifactIdQueryAsync(string workspaceNam
...
+		}
 
-					QueryResultSet<Workspace> workspaceQueryResultSet = await Task.Run(() => rsapiClient.Repositories.Workspace.Query(workspaceQuery));
+		#region Workspace Rest Helpers
 
-					if (!workspaceQueryResultSet.Success || workspaceQueryResultSet.Results == null)
-					{
-						throw new Exception("Failed to query Workspaces");
-					}
+		private async Task<List<int>> GetArtifactIDList(HttpClient httpClient, string url)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/WorkspaceHelper.cs:342]: - [ ] Update all async methods with async suffix

File: CSharp/DevVmPowershell/Helpers/Implementations/RestHelper.cs [hunk 2]
```diff
@@ -27,25 +27,25 @@ public static HttpClient GetHttpClient(string instanceAddress, string adminUsern
 			return httpClient;
 		}
 
-		public static HttpResponseMessage MakePost(HttpClient httpClient, string url, string request)
+		public static async Task<HttpResponseMessage> MakePostAsync(HttpClient httpClient, string url, string request)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/RestHelper.cs:30]: We should not have these methods as static. We are not adding unit tests now, but if in future we want to add unit tests, having static methods will not make it possible.

- [ ] Can we convert them to not static methods and do DI where ever RestHelper is needed.

**Review by calimetikcura — CHANGES_REQUESTED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 3]
```diff
@@ -140,7 +140,7 @@ private void CreateDisclaimerConfigurationRDO(int objectTypeId, int layoutId, in
 				}
 			};
 			string request = JsonConvert.SerializeObject(objectManagerCreateRequestModel);
-			HttpResponseMessage response = RestHelper.MakePost(httpClient, url, request);
+			HttpResponseMessage response = RestHelper.MakePostAsync(httpClient, url, request).Result;
```
[the remark this answers is not in the document]
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:143]: I can make the method async, but it is being called in the Powershell Module which is not async. So then I would just have to override the async by doing .Wait()

**Review by TommyBigNose — APPROVED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/ImportApiHelper.cs [hunk 4]
```diff
@@ -129,7 +129,8 @@ static void ImportJobOnComplete(JobReport jobReport)
 
 		static async Task<int> GetNumberOfDocumentsAsync(int workspaceId, string fileType)
 		{
-			HttpClient httpClient = RestHelper.GetHttpClient(InstanceAddress, AdminUsername, AdminPassword);
+			RestHelper restHelper = new RestHelper();
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/ImportApiHelper.cs:132]: - [ ] Pass RestHelper to the ImportApiHelper constructor and use it for making it DI

File: CSharp/DevVmPowershell/Helpers/Implementations/InstanceSettingsHelper.cs [hunk 5]
```diff
@@ -101,7 +104,8 @@ public bool UpdateInstanceSettingValue(string name, string section, string newVa
 				};
 
 				string queryRequest = JsonConvert.SerializeObject(objectManagerQueryRequestModel);
-				HttpResponseMessage queryResponse = RestHelper.MakePost(httpClient, Constants.Connection.RestUrlEndpoints.ObjectManager.queryUrl, queryRequest);
+				HttpResponseMessage queryResponse = await RestHelper.MakePostAsync(httpClient,
+					Constants.Connection.RestUrlEndpoints.ObjectManager.queryUrl, queryRequest);
 				if (!queryResponse.IsSuccessStatusCode)
 				{
 					throw new Exception("Failed to Query for Agent Artifact Ids");
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/InstanceSettingsHelper.cs:111]: - [ ] The comment is mentioning Agent when it should be InstanceSettings

**Review by calimetikcura — CHANGES_REQUESTED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/ImportApiHelper.cs (hunk 4 above)
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Implementations/ImportApiHelper.cs:132]: @calimetikcura The method this is in is a static method. So the RestHelper can not be passed into the constructor and used. Should I make the method non static?

Comment (calimetikcura) [reply CSharp/DevVmPowershell/Helpers/Implementations/ImportApiHelper.cs:132]: ah. that's fine for now as you have not touched that method.

**Review by calimetikcura — APPROVED** [review]

**Review by TommyBigNose — APPROVED** [review]
I have been merging with my branch for a bit and it's been working so far after adjusting for conflicts

**Pull request merged**